test/test_entry_filter.rb

```diff
@@ -149,5 +149,12 @@ class TestEntryFilter < JekyllUnitTest
       assert @filter.glob_include?(data, "vendor/bundle/jekyll/lib/page.rb")
       assert @filter.glob_include?(data, "/vendor/ruby/lib/set.rb")
     end
+
+    should "match directory only if there is trailing slash" do
+      data = ["../_glob_include_test/_is_dir/", "../_glob_include_test/_not_dir/"]
```

Comment:
our test site is in `test/source`. Please place the two `_glob_include_test` files in `test/source`.